### PR TITLE
docs(skills): add pr-comments skill for addressing PR review feedback

### PR DESCRIPTION
**Pull request by mcaden**
Adds a new repo skill for both Copilot CLI and Claude that walks through addressing GitHub PR review comments end-to-end.

## What the skill covers

- Discover open review threads via GraphQL (with `pageInfo` so truncation is detectable, and `pullRequestReview { state }` so PENDING comments can be filtered)
- Triage each thread into one of: `accept` / `decline` / `question` / `defer` / `already-done` / `outdated`
- Implement accepted changes; run the local quality gate (`npm run lint`, `npm test -- --run`, `cargo fmt`, `cargo clippy`, `cargo test`); never use `--no-verify`
- Push once per batch (capture `` for replies)
- Reply in-thread via REST `/comments/{databaseId}/replies` (or GraphQL `addPullRequestReviewThreadReply`) with the **mandatory** disclaimer prefix
- Resolve via GraphQL `resolveReviewThread` — **only** for the `accept` class

## Two load-bearing rules

1. **Every reply starts with** `🤖 AI agent reply (acting for @<gh-user>):` then a blank line then the body. No exceptions, including for one-line "done" replies. Tool-agnostic ("AI agent", not Copilot vs Claude).
2. **Resolve only when code was actually changed in response.** Questions, declines, deferrals, `already-done`, and `outdated` all stay open for the human.

## Notable subtleties baked into the skill

- **Sticky classifications**: triage class is locked at §3 snapshot time. GitHub flips `isOutdated: true` for every thread the agent successfully addresses (because the push moves the anchored line) — the skill explicitly says do **not** re-classify between push and resolve, otherwise every `accept` would silently become `outdated` and never get resolved.
- **Self-loop guard**: requires both `last_comment.author.login == \` AND the disclaimer prefix; prefix-only matching would misfire if a human quoted the prefix.
- **Partial-failure recovery**: if any reply API call fails post-push, stop further mutations and surface the unprocessed thread IDs. On rerun, detect existing AI replies citing the current `\` to avoid duplicates.

## Wiring

- `.github/skills/pr-comments/SKILL.md` — the skill itself; auto-discovered by Copilot CLI (mirrors `quality-workflow`).
- `CLAUDE.md` — new "Skills (read on demand)" section pointing Claude at the file, since Claude doesn't auto-discover skills.
- `.github/copilot-instructions.md` — restates the two load-bearing rules in the always-loaded principles, so even an agent that doesn't load the skill won't post an undisclosed reply or auto-resolve a declined thread.

## Process

- Rubber-duck pass before authoring caught: `outdated` auto-resolve was too fuzzy (now: never auto-resolve), missing `pageInfo` on GraphQL queries, weak self-loop guard (prefix-only), missing partial-failure handling, missing POSIX-shell note.
- User-spotted gap: post-push `isOutdated` flip would have caused every successfully-fixed thread to skip the resolve. Added explicit "classifications are sticky from triage" subsection.
- Code-review pass caught a broken-grammar/wrong-endpoint paragraph in the Suggestion blocks edge case (the cited `/reactions` endpoint is for emoji reactions, not suggestion application — there's no API for the latter). Rewritten.

Local gates green: pre-commit + pre-push (npm test + cargo test) ran on this commit.

### CODE REVIEW

File: .github/skills/pr-comments/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+belong to an unsubmitted review and shouldn't be replied to. Keep
+`isOutdated` threads in scope but flag them — the line they reference
+may no longer exist, which changes how you respond.
+
+If `reviewThreads.pageInfo.hasNextPage` or any per-thread
+`comments.pageInfo.hasNextPage` is `true`, **paginate** with the
+returned `endCursor` (add an `after: $cursor` variable). Don't proceed
+on a partial dataset — silently skipping threads is worse than asking
+the user to confirm the scope.
+
```
[the remark this answers is not in the document]
Comment (mcaden) [reply .github/skills/pr-comments/SKILL.md]: 🤖 AI agent reply (acting for @mcaden):

Done in cda3eb6. §3 pagination guidance now distinguishes the two connections explicitly: re-issue the outer query with `after: $cursor` for `reviewThreads` pagination, and for any thread whose `comments.pageInfo.hasNextPage` is true, walk that thread's connection in a *separate* per-thread query (`node(id: $threadId) { ... on PullRequestReviewThread { comments(first: 50, after: $commentCursor) {...} } }`). Cursor types don't match across connections, so a single global `after` won't work — the rewritten section calls that out.

File: .github/skills/pr-comments/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+3. **Quality gate** — lint/test/clippy locally; fix or revert before
+   pushing.
+4. **Push** — once per batch, after all `accept` changes for the PR are
+   committed (avoids one push per thread).
+5. **Reply** (§6) — disclaimer + body + commit ref.
+6. **Resolve** (§7) — only `accept`.
+
+Doing all the implementation work *before* any replies means: (a) the
+reply can cite a real, pushed SHA; (b) if the quality gate fails, no
+replies have been posted yet, so nothing to retract.
```
[the remark this answers is not in the document]
Comment (mcaden) [reply .github/skills/pr-comments/SKILL.md:345]: 🤖 AI agent reply (acting for @mcaden):

Done in cda3eb6. §5 now says `before pushing` (not `before committing`), which matches the §8 recipe order: implement → commit → quality gate → push. The pre-commit Husky hook handles per-staged-file lint at commit time; the full suite (`npm test -- --run` + `cargo test --workspace`) is the pre-push concern. Also added that fixup commits on the same branch are the right way to address gate failures (or amend if the broken commit hasn't been pushed yet) — `--no-verify` is still forbidden.

File: CLAUDE.md [hunk 3]
```diff
@@ -137,3 +137,10 @@ Rust backend owns all PTYs and persistent state; the React frontend communicates
 - Rust command handlers are always `async fn`, return `Result<T, AppError>` (AppError is `serde::Serialize`), and contain no business logic — delegate to `commands/session.rs`.
 - Event names use `://` namespace (e.g., `session://output`); payloads are always named-field structs.
 - Path alias `@/*` → `src/*`; no deep relative imports.
+
+## Skills (read on demand)
+
+Claude does not auto-discover skill files. When a task matches one of these, **read the linked file before acting**:
+
+- **Addressing PR review comments / responding to PR feedback** → read `.github/skills/pr-comments/SKILL.md`. Covers the exact `gh api` / GraphQL calls for listing review threads, replying in-thread (with the **mandatory `🤖 AI agent reply (acting for @<user>):` disclaimer prefix**), and the resolve-only-when-code-changed policy.
```
[the remark this answers is not in the document]
Comment (mcaden) [reply CLAUDE.md]: 🤖 AI agent reply (acting for @mcaden):

Done in cda3eb6. Standardized on `@<gh-user>` across all three docs (CLAUDE.md, .github/copilot-instructions.md, and SKILL.md §6) and explicitly stated in each that `<gh-user>` is replaced literally with the output of `gh api user --jq .login` (e.g. `@mcaden`). The `@<ME>` form in the SKILL.md disclaimer block was confusing because `ME` was a shell variable name, not a placeholder; it's now `@<gh-user>` to match.

File: .github/skills/pr-comments/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+  npm test -- --run
+  cargo fmt --all -- --check
+  cargo clippy --workspace --all-targets -- -D warnings
+  cargo test --workspace
+  ```
+
+  If any of these fail, fix in a follow-up commit on the same branch
+  (or amend if the broken commit hasn't been pushed yet) before
+  pushing — **never** bypass hooks with `--no-verify` (see
+  `.github/copilot-instructions.md` "Dogfooding safety").
```
[the remark this answers is not in the document]
Comment (mcaden) [reply .github/skills/pr-comments/SKILL.md]: 🤖 AI agent reply (acting for @mcaden):

Done in c6081f6. Updated the cross-reference to `Shift-left quality (principles)`, which is where the `--no-verify` rule actually lives in `.github/copilot-instructions.md`.

File: .github/skills/pr-comments/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+  user which PR to address — don't open a new one.
+- **Pending/draft reviews**: comments inside a `PENDING` review aren't
+  visible to others yet. Skip them; they'll show up once submitted.
+- **Agent's own previous AI replies (self-loop guard)**: when deciding
+  whether to reply again, look at the thread's **last** comment and
+  require **both** of:
+  - `comments.nodes[-1].author.login == $ME` (the comment was posted by
+    the same gh account the agent is using), **and**
+  - `comments.nodes[-1].body` starts with `🤖 AI agent reply` (matches
+    the disclaimer prefix exactly).
```
[the remark this answers is not in the document]
Comment (mcaden) [reply .github/skills/pr-comments/SKILL.md]: 🤖 AI agent reply (acting for @mcaden):

Done in c6081f6. Tightened the self-loop guard to match the literal disclaimer prefix up through `🤖 AI agent reply (acting for @` (the username and closing `):` are the only varying parts), and reframed the trailing rationale — with the author-login check already in the rule, the relevant edge case is the opposite (a same-account human comment without the disclaimer), not a human quoting the prefix.

File: .github/skills/pr-comments/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+
+### Posting the reply
+
+Reply **inside the existing thread** (not as a new top-level review
+comment) so the conversation stays threaded:
+
+```sh
+# $TOP_COMMENT_ID is the databaseId of the *first* comment in the thread
+gh api -X POST \
+  "repos/$OWNER/$REPO/pulls/$PR/comments/$TOP_COMMENT_ID/replies" \
```
[the remark this answers is not in the document]
Comment (mcaden) [reply .github/skills/pr-comments/SKILL.md:278]: 🤖 AI agent reply (acting for @mcaden):

The documented path is correct as written. Per the official REST API reference (https://docs.github.com/en/rest/pulls/comments?apiVersion=2022-11-28#create-a-reply-for-a-review-comment), the endpoint for `Create a reply for a review comment` is:

`POST /repos/{owner}/{repo}/pulls/{pull_number}/comments/{comment_id}/replies`

The `{pull_number}` segment is required — the docs' own cURL example is `…/pulls/PULL_NUMBER/comments/COMMENT_ID/replies`. Empirically, the two replies I just posted on this PR (including this one) used exactly the path the skill documents (`repos///pulls//comments//replies`) and returned 201, not 404. The path that omits `{pull_number}` belongs to a different endpoint (`GET/PATCH/DELETE` of a single review comment), which is likely the source of the confusion.
